symbulate/results.py

```diff
@@ -313,78 +337,96 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
-                res = pd.DataFrame({'X': x, 'Y': y})
-                res['num'] = 1
-                temp = pd.pivot_table(res, values = 'num', index = ['Y'],
-                    columns = ['X'], aggfunc = np.sum)
-                temp = temp / len(x)
-                sns.set()
-                sns.cubehelix_palette(8)
-                fig, ax = plt.subplots(1, 1)
-                sns.heatmap(temp, ax=ax, linewidths = 0.03).invert_yaxis()
-                ax.set_ylabel('')
-                ax.set_xlabel('')
-            elif type == "mosaic" and discrete_x and discrete_y:
-                res = pd.DataFrame({'X': x, 'Y': y})
-                ct = pd.crosstab(res['Y'], res['X'])
-                ctplus = ct + 1e-8
-                labels = lambda k: ""
-                fig, ax = plt.subplots(1, 1)
-                temp = mosaic(ctplus.unstack(), ax = ax, labelizer = labels, axes_label = False)
-            elif discrete_x and discrete_y:
-                raise Exception("Must have type='mosaic', 'tile', or 'scatter' if discrete.")
-            elif discrete_x == True and discrete_y == False and type == 'violin':
-                fig, ax = plt.subplots(1, 1)
-                res = pd.DataFrame({'X': x, 'Y': y})
-                sns.violinplot(x='X', y='Y', data=res)
-                ax.set_ylabel('')
-                ax.set_xlabel('')
-            elif discrete_x == False and discrete_y == True and type == 'violin':
-                #TODO change this situation based on Dr. Ross's suggestions
+            nullfmt = NullFormatter() #removes labels on fig
+
+            if 'scatter' in type:
                 if jitter:
                     x += np.random.normal(loc=0, scale=.01 * (max(x) - min(x)), size=len(x))
                     y += np.random.normal(loc=0, scale=.01 * (max(y) - min(y)), size=len(y))
-                # get next color in cycle
-                axes = plt.gca()
-                color = get_next_color(axes)
-                plt.scatter(x, y, color=color, alpha=alpha, **kwargs)
-            else:
-                raise Exception("I don't know how to plot these variables.")
+                ax.scatter(x, y, alpha=alpha, c=color, **kwargs)
+            elif 'hist' in type:
+                histo = ax.hist2d(x, y, bins=bins, cmap='Blues')
+                cbar, caxes = add_colorbar(fig, type, histo[3], 'Density')
+                #change scale to density instead of counts
+                new_labels = []
+                for label in caxes.get_yticklabels():
+                    new_labels.append(int(label.get_text()) / len(x))
+                caxes.set_yticklabels(new_labels)
+            elif 'density' in type:
+                den = make_density2D(x, y, ax)
+                cbar, caxes = add_colorbar(fig, type, den, 'Density')
+            elif 'tile' in type:
+                if not discrete_x and not discrete_y:
+                   raise Exception("type='tile' is only valid for 2 discrete variables" )
```

Comment:
That's not true, is it? It seems like your tile plot will work if there's 1 discrete and 1 continuous, or even if both variables are continuous. If so, you may just want to abandon this error message altogether.